app/main.py

```diff
@@ -1,3 +1,30 @@
-def shop_trip():
-    # write your code here
-    pass
+import json
+from typing import Callable
+
+from app.shop import Shop
+from app.customer import Customer
+
+
+def get_data(info_key: str, creation_func: Callable) -> list:
+    with open("app/config.json", "r") as source:
+        data = json.load(source)
+    return [creation_func(info) for info in data[info_key]]
```

Comment:
To prevent `KeyError` exceptions, consider checking if `info_key` exists in `data` before accessing it. You can use `data.get(info_key, [])` to provide a default empty list if the key is missing. This will make the code more robust against missing keys.